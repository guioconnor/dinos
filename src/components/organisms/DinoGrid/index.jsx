import React from "react";
import styled from "styled-components";
import DinoLink from "../../atoms/DinoLink";
import DinoCard from "../DinoCard";
import Filter from "../../../containers/FilterContainer";

const dietFilterValues = [
  { value: null, name: "All Diets" },
  { value: "carnivore", name: "Carnivore" },
  { value: "herbivore", name: "Herbivore" },
  { value: "omnivore", name: "Omnivore" }
];

const periodFilterValues = [
  { value: null, name: "All Periods" },
  { value: "Early Jurassic", name: "Early Jurassic" },
  { value: "Late Jurassic", name: "Late Jurassic" },
  { value: "Early Cretaceous", name: "Early Cretaceous" },
  { value: "Late Cretaceous", name: "Late Cretaceous" }
];

const Container = styled.ul`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
`;

const GridItem = styled.li`margin: 20px;`;

const StyledDinoLink = styled(DinoLink)`
  display: block;
  margin: 20px;
`;

const H1 = styled.h1`font-size: 50px;`;

const DinoGrid = ({ dinoData, mainIllustration }) => (
  <div className="App">
    <H1>Dinosaurs</H1>
    <Filter filterName="diet" filterValues={dietFilterValues} />
    <Filter filterName="period" filterValues={periodFilterValues} />
    <Container>
      {dinoData.map(dino => (
        <GridItem>
          <StyledDinoLink dinoName={dino.name}>
            <DinoCard dino={dino} displayDetails />
          </StyledDinoLink>
        </GridItem>
      ))}
    </Container>
  </div>
);

export default DinoGrid;
