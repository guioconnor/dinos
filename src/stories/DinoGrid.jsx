import React from "react";
import { storiesOf } from "@storybook/react";
import DinoGrid from "../components/organisms/DinoGrid";

const dinoData = [
  {
    // image: Silhouettes.allosaurus,
    name: "Allosaurus",
    pronunciation: "AL-oh-sore-us",
    meaningOfName: "other lizard",
    diet: "carnivorous",
    length: "12m",
    period: "Late Jurassic",
    mya: "156-144",
    info:
      "Allosaurus was an apex predator in the Late Jurassic in North America."
  },
  {
    // "image": apatosaurus,
    name: "Apatosaurus",
    pronunciation: "ah-PAT-oh-sore-us",
    meaningOfName: "deceptive lizard",
    diet: "herbivorous",
    length: "21m",
    period: "Late Jurassic",
    mya: "154-145",
    info:
      "Named the 'deceptive lizard' because its skull was confused with those of other sauropods until 1909."
  }
];

storiesOf("DinoGrid", module).add("default", () => (
  <DinoGrid dinoData={dinoData} mainIllustration="stub" />
));
