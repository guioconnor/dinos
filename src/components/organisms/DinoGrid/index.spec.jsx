import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import DinoGrid from "./index";

const data = [
  {
    image: "Silhouettes.allosaurus",
    name: "Allosaurus",
    pronunciation: "AL-oh-sore-us",
    meaningOfName: "other lizard",
    diet: "carnivorous",
    length: "12m",
    period: "Late Jurassic",
    mya: "156-144",
    info:
      "Allosaurus was an apex predator in the Late Jurassic in North America."
  },
  {
    name: "Apatosaurus",
    pronunciation: "ah-PAT-oh-sore-us",
    meaningOfName: "deceptive lizard",
    diet: "herbivorous",
    length: "21m",
    period: "Late Jurassic",
    mya: "154-145",
    info:
      "Named the 'deceptive lizard' because its skull was confused with those of other sauropods until 1909."
  }
];

describe("<DinoGrid />", () => {
  describe("With an empty data set", () => {
    const wrapper = shallow(<DinoGrid dinoData={[]} />);

    it("should render an empty grid", () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe("With a non-empty data set", () => {
    const wrapper = shallow(<DinoGrid dinoData={data} />);

    it("should render a grid with dino cards", () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
