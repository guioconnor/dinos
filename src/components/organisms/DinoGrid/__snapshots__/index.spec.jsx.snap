// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DinoGrid /> With a non-empty data set should render a grid with dino cards 1`] = `
<div
  className="App"
>
  <styled.h1>
    Dinosaurs
  </styled.h1>
  <Connect(PeriodFilter)
    filterName="diet"
    filterValues={
      Array [
        Object {
          "name": "All Diets",
          "value": null,
        },
        Object {
          "name": "Carnivore",
          "value": "carnivore",
        },
        Object {
          "name": "Herbivore",
          "value": "herbivore",
        },
        Object {
          "name": "Omnivore",
          "value": "omnivore",
        },
      ]
    }
  />
  <Connect(PeriodFilter)
    filterName="period"
    filterValues={
      Array [
        Object {
          "name": "All Periods",
          "value": null,
        },
        Object {
          "name": "Early Jurassic",
          "value": "Early Jurassic",
        },
        Object {
          "name": "Late Jurassic",
          "value": "Late Jurassic",
        },
        Object {
          "name": "Early Cretaceous",
          "value": "Early Cretaceous",
        },
        Object {
          "name": "Late Cretaceous",
          "value": "Late Cretaceous",
        },
      ]
    }
  />
  <styled.div>
    <Styled(DinoLink)
      dinoName="Allosaurus"
    >
      <DinoCard
        dino={
          Object {
            "diet": "carnivorous",
            "image": "Silhouettes.allosaurus",
            "info": "Allosaurus was an apex predator in the Late Jurassic in North America.",
            "length": "12m",
            "meaningOfName": "other lizard",
            "mya": "156-144",
            "name": "Allosaurus",
            "period": "Late Jurassic",
            "pronunciation": "AL-oh-sore-us",
          }
        }
        displayDetails={true}
      />
    </Styled(DinoLink)>
    <Styled(DinoLink)
      dinoName="Apatosaurus"
    >
      <DinoCard
        dino={
          Object {
            "diet": "herbivorous",
            "info": "Named the 'deceptive lizard' because its skull was confused with those of other sauropods until 1909.",
            "length": "21m",
            "meaningOfName": "deceptive lizard",
            "mya": "154-145",
            "name": "Apatosaurus",
            "period": "Late Jurassic",
            "pronunciation": "ah-PAT-oh-sore-us",
          }
        }
        displayDetails={true}
      />
    </Styled(DinoLink)>
  </styled.div>
</div>
`;

exports[`<DinoGrid /> With an empty data set should render an empty grid 1`] = `
<div
  className="App"
>
  <styled.h1>
    Dinosaurs
  </styled.h1>
  <Connect(PeriodFilter)
    filterName="diet"
    filterValues={
      Array [
        Object {
          "name": "All Diets",
          "value": null,
        },
        Object {
          "name": "Carnivore",
          "value": "carnivore",
        },
        Object {
          "name": "Herbivore",
          "value": "herbivore",
        },
        Object {
          "name": "Omnivore",
          "value": "omnivore",
        },
      ]
    }
  />
  <Connect(PeriodFilter)
    filterName="period"
    filterValues={
      Array [
        Object {
          "name": "All Periods",
          "value": null,
        },
        Object {
          "name": "Early Jurassic",
          "value": "Early Jurassic",
        },
        Object {
          "name": "Late Jurassic",
          "value": "Late Jurassic",
        },
        Object {
          "name": "Early Cretaceous",
          "value": "Early Cretaceous",
        },
        Object {
          "name": "Late Cretaceous",
          "value": "Late Cretaceous",
        },
      ]
    }
  />
  <styled.div />
</div>
`;
